libdevcore/StringUtils.h

```diff
@@ -72,4 +75,78 @@ std::string joinHumanReadable
 	return result;
 }
 
+/// Returns decimal representation for smaller numbers; hex for large numbers.
+/// "Special" numbers, powers-of-two and powers-of-two minus 1, are given in
+/// formulaic form like 0x01 * 2**24 - 1.
+/// If the _useTruncation parameter is true, internal truncation is also applied,
+/// like  0x5555...{+56 more}...5555
+/// @a T will typically by unsigned, u160, u256 or bigint.
+	template <class T>
+	inline std::string formatNumberReadable(
+			T _value,
+			bool _useTruncation = false
+	)
+	{
+		static_assert(
+				std::is_same<bigint, T>::value || !std::numeric_limits<T>::is_signed,
+				"only unsigned types or bigint supported"
+		); //bigint does not carry sign bit on shift
+
+		// smaller numbers return as decimal
+		if (_value <= 0x1000000)
+			return _value.str();
+
+		HexCase hexcase = HexCase::Mixed;
+		HexPrefix prefix = HexPrefix::Add;
+
+		// when multiple trailing zero bytes, format as N * 2**x
+		int i;
+		T v = _value;
+		for (i = 0; (v & (T)0xff) == 0; i++, v >>= 8)
```

Comment:
Please don't use C-style casts and don't use the comma operator. The `v>>=8` can easily move into the loop body.